be/src/runtime/exec_env.cpp

```diff
@@ -410,79 +420,97 @@ Status CacheEnv::_init_datacache() {
 #endif
 
     if (config::datacache_enable) {
-        CacheOptions cache_options;
-        RETURN_IF_ERROR(DataCacheUtils::parse_conf_datacache_mem_size(
-                config::datacache_mem_size, _global_env->process_mem_limit(), &cache_options.mem_space_size));
-
-        for (auto& root_path : _store_paths) {
-            // Because we have unified the datacache between datalake and starlet, we also need to unify the
-            // cache path and quota.
-            // To reuse the old cache data in `starlet_cache` directory, we try to rename it to the new `datacache`
-            // directory if it exists. To avoid the risk of cross disk renaming of a large amount of cached data,
-            // we do not automatically rename it when the source and destination directories are on different disks.
-            // In this case, users should manually remount the directories and restart them.
-            std::string datacache_path = root_path.path + "/datacache";
-            std::string starlet_cache_path = root_path.path + "/starlet_cache/star_cache";
-#ifdef USE_STAROS
-            if (config::datacache_unified_instance_enable) {
-                RETURN_IF_ERROR(DataCacheUtils::change_disk_path(starlet_cache_path, datacache_path));
-            }
+#if defined(WITH_STARCACHE)
+        ASSIGN_OR_RETURN(auto cache_options, _init_cache_options());
+
+        // init starcache & disk monitor
+        _local_cache = std::make_shared<StarCacheWrapper>();
+        _disk_space_monitor = std::make_shared<DiskSpaceMonitor>(_local_cache.get());
+        RETURN_IF_ERROR(_disk_space_monitor->init(&cache_options.disk_spaces));
+        RETURN_IF_ERROR(_local_cache->init(cache_options));
+        _disk_space_monitor->start();
```

Comment:
Why not move the logic related to the `disk_space_monitor` to the local cache ?

Reply:
The monitor needs to manage the entire disk quota, and coordinate the disk space of the spill, internal tables, and local cache later. We should separate the background threads from the cache class.